pkgs/servers/monitoring/prometheus/alertmanager.nix

```diff
@@ -1,33 +1,28 @@
-{ lib, go, buildGoPackage, fetchFromGitHub, installShellFiles }:
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "alertmanager";
-  version = "0.21.0";
+  version = "0.23.0";
   rev = "v${version}";
 
-  goPackagePath = "github.com/prometheus/alertmanager";
-
   src = fetchFromGitHub {
     inherit rev;
     owner = "prometheus";
     repo = "alertmanager";
-    sha256 = "0zrzyaqs73pz4rmj4xaj15x4n1542m0nb7jqm2j77k07j75r5w41";
+    sha256 = "sha256-06mKgWUyw5jsjKiRXQ9/oqHvFdkY2nS9Z3eW60lTNbU=";
   };
 
-  ldflags = let t = "${goPackagePath}/vendor/github.com/prometheus/common/version"; in [
-    "-X ${t}.Version=${version}"
-    "-X ${t}.Revision=${src.rev}"
-    "-X ${t}.Branch=unknown"
-    "-X ${t}.BuildUser=nix@nixpkgs"
-    "-X ${t}.BuildDate=unknown"
-    "-X ${t}.GoVersion=${lib.getVersion go}"
-  ];
```

Comment:
This was needed to display useful information for `--version`, so I think we'll still need it somehow.

Reply:
Yeah, forgot about adding this again. Fixed. Thanks!